### ISSUE TITLE
Data management workflows

### ISSUE THREAD

**Issue by tidoust** [ddbeck/common-web-feature-mockup#1]
The workflows that will be applied to data to get to the final data file are likely going to affect the structure of these data files. These workflows will probably involve a combination of automated jobs and manual reviews and updates.

For instance, finding constituent features of a feature can be semi-automated provided the data includes "starting points", e.g. the URL of one or more specs that define the feature (or of a section in a spec), perhaps a list of interfaces, etc. Starting points may not need to appear in the final data file.

Similarly, when someone adds a constituent feature manually, we may need to flag it in a special way (or flag those that are suggested/maintained automatically) so that a job that runs on the data does not try to touch the entry.

I think it would be useful to sketch these workflows early on, with a view to maximizing the number of automatic steps!

**Comment by ddbeck** [comment]
> I think it would be useful to sketch these workflows early on, with a view to maximizing the number of automatic steps!

@tidoust Yes! I think some of these workflows will depend on what taxonomy (taxonomies?) we choose for our feature grouping. For instance, our approach to workflow might look different if we were to build up features with numbered versions (e.g., WebRTC 1, WebRTC 2, and so on) versus annualized versions (2022, 2023…). So I don't know what exactly we might encode in terms of starting points yet.

But some are going to really obvious. For example, a snapshot (e.g., CSS 2023), historic group (CSS2, for instance ), or externally-sourced group (e.g., ECMAScript 2021) might never change. So we ought to have a way of indicating that a group is "frozen". That seems like a great place to start annotating these things. But I imagine it won't be the only sort of annotation we need.

This provoked a lot of thinking on how we might signal amongst ourselves the progression of various features and their ownership. I opened https://github.com/ddbeck/common-web-feature-mockup/issues/5 with some ideas on we might encode sustainable review and maintenance practices into the authorship of the groups themselves. I still think there's a lot of utility in sketching out the workflows, so I don't think this issue supersedes that one, but it does touch on this (and, to a degree, #3).
